### PR TITLE
complete react-events-and-state

### PR DESCRIPTION
**Pull request by hoangnguyen9384**
Screen Shot 2021-03-17 at 8 10 15 AM

### CODE REVIEW

File: react-events-and-state/src/index.jsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+class CustomButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { isClicked: false };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    this.setState(state => ({ isClicked: true }));
```
Comment (scbowler) [inline react-events-and-state/src/index.jsx:12]: For most cases you don't need to pass `setState` a callback function, you can just pass it an object with the properties you want to update.

```JSX
this.setState({ isClicked: true });
```

File: react-events-and-state/src/index.jsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  }
+
+  render() {
+    return (
+     <button onClick={this.handleClick}>
+      {this.state.isClicked ? 'Thanks!' : 'Click Me!'}
+     </button>
```
Comment (scbowler) [inline react-events-and-state/src/index.jsx:19]: Great job here!

**Review by scbowler — APPROVED** [review]
Well done! 👍

File: react-events-and-state/src/index.jsx (hunk 1 above)
Comment (hoangnguyen9384) [reply react-events-and-state/src/index.jsx:12]: I got it! Thanks!

**Pull request merged**